source/auth/tests/mongodb-oidc.rst

```diff
@@ -39,149 +38,65 @@ is one principal configured.
 
 Single Principal Implicit Username
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a request callback returns a valid token.
-- Create a client that uses the default OIDC url and the request callback.
+- Create default OIDC client.
 - Perform a ``find`` operation. that succeeds.
 - Close the client.
 
 Single Principal Explicit Username
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a request callback that returns a valid token.
-- Create a client with a url of the form  ``mongodb://test_user1@localhost/?authMechanism=MONGODB-OIDC`` and the OIDC request callback.
+- Create a client with ``MONGODB_URI_SINGLE``, a username of ``test_user1``, and the OIDC request callback.
 - Perform a ``find`` operation that succeeds.
 - Close the client.
 
 Multiple Principal User 1
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a request callback that returns a valid token.
-- Create a client with a url of the form  ``mongodb://test_user1@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Create a client with ``MONGODB_URI_MULTI``, a username of ``test_user1``, and the OIDC request callback.
 - Perform a ``find`` operation that succeeds.
 - Close the client.
 
 Multiple Principal User 2
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
 - Create a request callback that reads in the generated ``test_user2`` token file.
-- Create a client with a url of the form ``mongodb://test_user2@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Create a client with ``MONGODB_URI_MULTI``, a username of ``test_user2``, and the OIDC request callback.
 - Perform a ``find`` operation that succeeds.
 - Close the client.
 
 Multiple Principal No User
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a client with a url of the form ``mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Create a client with ``MONGODB_URI_MULTI``, no username, and the OIDC request callback.
 - Assert that a ``find`` operation fails.
 - Close the client.
 
 Allowed Hosts Blocked
 ~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a client that uses the OIDC url and a request callback, and an
-  ``ALLOWED_HOSTS`` that is an empty list.
+- Create a default OIDC client, with an ``ALLOWED_HOSTS`` that is an empty list.
```

Comment:
Could we either split this into two files, or into two distinct sections, one for machine workflow tests, and the other for human workflow tests (which can all be omitted, for drivers that do not implement them)?

Reply:
Done!